editor/src/components/navigator/navigator-item/item-label.tsx

```diff
@@ -245,9 +282,24 @@ export const ItemLabel = React.memo((props: ItemLabelProps) => {
             }
           }}
         >
-          {value}
+          {label}
         </div>
       )}
     </div>
   )
 })
+
+function remixSceneLocationFromOutletPath(
+  outletPath: ElementPath,
+  remixNavigationData: RemixNavigationAtomData,
+): Location | null {
+  let parentPath = EP.dropNPathParts(outletPath, 1)
+  while (!EP.pathsEqual(parentPath, EP.emptyElementPath)) {
+    const maybeRemixData = remixNavigationData[EP.toString(parentPath)]
+    if (maybeRemixData != null) {
+      return maybeRemixData.location
+    }
+    parentPath = EP.dropNPathParts(parentPath, 1)
+  }
```

Comment:
You've used this pattern twice in this PR, and I'd say it's worth refactoring part of it into a function in `element-path.ts`. We have the existing function `getAncestors` which will return the paths of all ancestors of a given path, and it will be ordered from longest path to shortest. You could create a similar function that is returning all of the paths of the "instances" (I can't think of a better term here), and then use `.find()` on the result of that here

Reply:
how about this? https://github.com/concrete-utopia/utopia/pull/4249/commits/f33df4d26bef5b19f66bd6054ea32edd9ece37c9

Reply:
and it's used like this https://github.com/concrete-utopia/utopia/pull/4249/commits/37afe0342eb3dc1fe927449208012b575436678c